### DISCUSSION TITLE
Easiest way to get data from custom Qdialog child component back to parent?

### DISCUSSION THREAD

**Discussion by NomanTrips** [quasarframework/quasar#13325] [General - Components / Directives / etc] [answered]
I'm implementing a quasar custom Qdialog component which is just a pop-up that you get to from a button click. I can't figure out from the docs or other online resources how to get data manipulated on the dialog back to the parent. I realize this is more of a vue thing but trying those approaches isn't getting me anywhere (parent listeners don't seem to hear anything).

I've tried:

1. this.$parent.$emit('clicked', val); from the child component. (don't know exactly where to put listener for this on the parent, most vue examples I've seen have it in the template, but in Qdialog case, no custom element in the template to put the listener in)
3. onDialogOK in the setup hook on the dialog component. (can't seem to access data or computed from here, unsure if wrong syntax or by design.)

**Comment by pdanpdan** [comment]
If you want to process data at dialog close then pass an object with what you want to onOk and use it in the opener
If you need to pass things while the dialog is open then you need to use something like a store

**Reply by NomanTrips** [reply]
Thanks for that, I did try passing the data onOk but I had trouble finding examples of how to do that, and all I could get back was a string or object I hard coded in there. My difficulty was in getting a computed value out of the child component via the onOk. Doing things such as this.value etc. resulted in an undefined. I'm pretty sure I'm not using this nifty new 'setup' hook correctly but don't know enough about vue 3 to really know what I'm doing wrong.

**Comment by pdanpdan** [comment, marked as answer]
make a codepen with what you have tried so we can help

**Reply by NomanTrips** [reply]
Appreciate the offer. I had trouble getting code pen to do multiple js files which I guess codesandbox.io is more for (I would kinda need multiple js files to demonstrate the problem).

So I found another solution. It seems I was not reading the documentation correctly and using the wrong dialog in quasar doc for what I'm trying to do.

What I did is:
**Use this:**
https://quasar.dev/vue-components/dialog#various-content
**instead of:**
https://quasar.dev/quasar-plugins/dialog#invoking-custom-component
